src/types/handnode.cc

```diff
@@ -71,14 +71,8 @@ bool Node::operator!=(const Node& n) const {
          leafPosInParent == n.leafPosInParent;
 }
 
-Node::~Node() {
-  if (parent != nullptr) {
-    parent->leaves.erase(parent->leaves.begin() + leafPosInParent);
-    for (size_t i = 0; i < parent->leaves.size(); i++) {
-      parent->leaves.at(i)->leafPosInParent = i;
-    }
-  }
-}
```

Comment:
This shouldn't free the parent, this should remove the reference to the current node in the parents leaves.

Reply:
Hmm I feel like we probably still need this the way that handnode is implemented, but I'll just request you to comment out the destructor and I'll address it in a follow up pr.